README.md

```diff
@@ -30,43 +30,79 @@ Install using your favorite plugin manager:
-
-          -- Path to the dbt project, if blank, will auto-detect
-          -- using currently open buffer for all sql,yml, and md files
-          path_to_dbt_project = "",
+        local dbt = require("dbtpal")
+        dbt.setup({
+            -- Path to the dbt executable
+            path_to_dbt = "dbt",
 
-          -- Path to dbt profiles directory
-          path_to_dbt_profiles_dir = vim.fn.expand "~/.dbt",
+            -- Path to the dbt project, if blank, will auto-detect
+            -- using currently open buffer for all sql,yml, and md files
+            path_to_dbt_project = "",
 
-          -- Search for ref/source files in macros and models folders
-          extended_path_search = true,
+            -- Path to dbt profiles directory
+            path_to_dbt_profiles_dir = vim.fn.expand("~/.dbt"),
 
-          -- Prevent modifying sql files in target/(compiled|run) folders
-          protect_compiled_files = true
+            -- Search for ref/source files in macros and models folders
+            extended_path_search = true,
 
-        }
+            -- Prevent modifying sql files in target/(compiled|run) folders
+            protect_compiled_files = true,
+        })
 
         -- Setup key mappings
 
-        vim.keymap.set('n', '<leader>drf', dbt.run)
-        vim.keymap.set('n', '<leader>drp', dbt.run_all)
-        vim.keymap.set('n', '<leader>dtf', dbt.test)
-        vim.keymap.set('n', '<leader>dm', require('dbtpal.telescope').dbt_picker)
+        vim.keymap.set("n", "<leader>drf", dbt.run)
+        vim.keymap.set("n", "<leader>drp", dbt.run_all)
+        vim.keymap.set("n", "<leader>dtf", dbt.test)
+        vim.keymap.set("n", "<leader>dm", require("dbtpal.telescope").dbt_picker)
 
         -- Enable Telescope Extension
-        require'telescope'.load_extension('dbtpal')
-        end,
-    requires = { { 'nvim-lua/plenary.nvim' }, {'nvim-telescope/telescope.nvim'} }
-    }
+        require("telescope").load_extension("dbtpal")
+    end,
+    requires = { { "nvim-lua/plenary.nvim" }, { "nvim-telescope/telescope.nvim" } },
+}
+```
+
+**Using lazy.nvim**
 
+<details>
+<summary>Show configuration...</summary>
+
+```lua
+{
+    "PedramNavid/dbtpal",
+    dependencies = {
+        "nvim-lua/plenary.nvim",
+        "nvim-telescope/telescope.nvim",
+    },
+    ft = {
+        "sql",
+        "md",
+        "yaml",
+    },
+    keys = {
+        { "<leader>drf", "<cmd>DbtRun<cr>" },
+        { "<leader>drp", "<cmd>DbtRunAll<cr>" },
+        { "<leader>dtf", "<cmd>DbtTest<cr>" },
+        { "<leader>dm", "<cmd>lua require('dbtpal.telescope').dbt_picker()<cr>" },
+    },
+    config = function()
+        require("dbtpal").setup({
+            path_to_dbt = "dbt",
+            path_to_dbt_project = "",
+            path_to_dbt_profiles_dir = vim.fn.expand("~/.dbt"),
+            extended_path_search = true,
+            protect_compiled_files = true,
+        })
+        require("telescope").load_extension("dbtpal")
```

Comment:
Side note, if we didn't have `require("telescope").load_extension("dbtpal")` we would be able to move all of the `setup` parameters to the _lazy.nvim_ opts parameter. Not sure how to get around loading the extension though.

Reply:
I think it's fine having it here right? Seems to work okay, not sure if there's any downsides.

Reply:
Yeah, I think it's ok!